test/integration/sync_with_devto_test.rb

```diff
@@ -0,0 +1,203 @@
+    FileUtils.mkdir_p(WORKING_DIR) # to avoid Errno::ENOENT
+  end
+
+  def run_sync
+    Sync.perform(http_client: FakeHttpClient.new, working_dir: WORKING_DIR)
+  end
+
+  def teardown
+    FileUtils.rm_rf(Dir.glob(WORKING_DIR))
+  end
+
+  def prepare_sync_file_with_un_synced_article
+    articles = {
+      1879395 => {
+        edited_at: "2023-10-23T00:00:00Z",
+        slug: "recent-searches-sorting-hashes-how-they-are-connected-ruby-rails",
+        synced: false
+      },
+      1877720 => {
+        edited_at: "2024-10-23T15:44:11Z",
+        slug: "myth-or-reality-can-test-driven-development-in-agile-replace-qa-programming",
+        synced: true
+      }
+    }
+    File.write("#{WORKING_DIR}#{SYNC_STATUS_FILE}", articles.to_yaml)
+  end
+
+  def sync_file_content
+    YAML.safe_load_file("#{WORKING_DIR}#{SYNC_STATUS_FILE}", permitted_classes: [Time, Symbol])
+  end
+
+  def change_article_slug
+    articles = {
+      1879395 => {
+        edited_at: "2023-10-23T00:00:00Z",
+        slug: "changed_slug",
+        synced: false
+      },
+      1877720 => {
+        edited_at: "2024-10-23T15:44:11Z",
+        slug: "myth-or-reality-can-test-driven-development-in-agile-replace-qa-programming",
+        synced: true
+      }
+    }
+    File.write("#{WORKING_DIR}#{SYNC_STATUS_FILE}", articles.to_yaml)
+  end
+
+  def test_sync_script_create_yaml_status_file_on_first_run
+    refute File.exist?("#{WORKING_DIR}#{SYNC_STATUS_FILE}")
+    run_sync
+    assert File.exist?("#{WORKING_DIR}#{SYNC_STATUS_FILE}")
+  end
+
+  def test_sync_script_fill_yaml_status_file
+    run_sync
+
+    assert_equal sync_file_content[1879395], {
+      edited_at: "2024-10-23T15:44:11Z",
+      slug: "recent-searches-sorting-hashes-how-they-are-connected-ruby-rails",
+      synced: true
+    }
+
+    assert_equal sync_file_content[1877720], {
+      edited_at: "2024-10-23T15:44:11Z",
+      slug: "myth-or-reality-can-test-driven-development-in-agile-replace-qa-programming",
+      synced: true
+    }
+  end
+
+  def test_sync_only_unsynced_articles
+    prepare_sync_file_with_un_synced_article
+
+    assert_equal sync_file_content, {
+      1879395 => {
+        edited_at: "2023-10-23T00:00:00Z",
+        slug: "recent-searches-sorting-hashes-how-they-are-connected-ruby-rails",
+        synced: false
+      },
+      1877720 => {
+        edited_at: "2024-10-23T15:44:11Z",
+        slug: "myth-or-reality-can-test-driven-development-in-agile-replace-qa-programming",
+        synced: true
+      }
+    }
+
+    run_sync
+
+    assert_equal sync_file_content[1879395], {
+        edited_at: "2024-10-23T15:44:11Z",
+        slug: "recent-searches-sorting-hashes-how-they-are-connected-ruby-rails",
+        synced: true
+      }
+
+    assert_equal sync_file_content[1877720], {
+        edited_at: "2024-10-23T15:44:11Z",
+        slug: "myth-or-reality-can-test-driven-development-in-agile-replace-qa-programming",
+        synced: true
+    }
+  end
+
```

Comment:
@dgorodnichy still has time changes without content updates. Please add a test that there is no dev.to API calls if there are no changes

![jetthoughts github io – Commit: index md 2024-11-12 13-40-16](https://github.com/user-attachments/assets/91954d0e-dc0b-4f83-ab8a-c23b487a815d)